go/interpreter/lfvm/lfvm_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package lfvm
+
+import (
+	"bytes"
+	"errors"
+	"testing"
+
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+)
+
+func TestVm_Run(t *testing.T) {
+
+	tests := map[string]struct {
+		code                  []byte
+		codeHash              tosca.Hash
+		revision              tosca.Revision
+		withSuperInstructions bool
+		withCodeCache         bool
+		expectedResult        tosca.Result
+		expectedError         *tosca.ErrUnsupportedRevision
+	}{
+		"empty code": {
+			revision: tosca.R13_Cancun,
+			expectedResult: tosca.Result{Success: true, GasLeft: 1000000,
+				GasRefund: 0, Output: []byte{}},
+		},
+		"invalid code": {
+			code:     []byte{0x0C},
+			revision: tosca.R13_Cancun,
+			expectedResult: tosca.Result{Success: false, GasLeft: 0,
+				GasRefund: 0, Output: []byte{}},
+		},
+		"newer unsupported revision": {
+			revision: newestSupportedRevision + 1,
+			expectedError: &tosca.ErrUnsupportedRevision{
+				Revision: newestSupportedRevision + 1},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+
+			params := tosca.Parameters{
+				Gas:      1000000,
+				Code:     test.code,
+				CodeHash: &test.codeHash,
+				BlockParameters: tosca.BlockParameters{
+					Revision: test.revision,
+				},
+			}
+
+			vm := &VM{
+				with_super_instructions: test.withSuperInstructions,
+				no_code_cache:           !test.withCodeCache,
+			}
+
+			result, err := vm.Run(params)
+			if err != nil || test.expectedError != nil {
```

Comment:
This code is cryptic, I think the new pattern with the error constructor function is better than the error type + `errors.As`
The new method allows us to use `errors.Is without checking for nil before in the both branches: yields straight forward code:

```go
   expecedErr := unsportedRevision(x) // or nil
   // ....
   if !errors.Is(err, expectedErr) { 
        // this branch satisfies: 
        //  - err is nil && expectedErr is not
        //  - err is not nil && err is != expectedErr
   }
   // jump is not taken if both are nil !
```